rustls/src/msgs/message.rs

```diff
@@ -368,16 +381,38 @@ impl Message {
             payload: MessagePayload::handshake(HandshakeMessagePayload::build_key_update_notify()),
         }
     }
+
+    pub(crate) fn into_owned(self) -> Message<'static> {
+        let Self { version, payload } = self;
+        Message {
+            version,
+            payload: payload.into_owned(),
+        }
+    }
+}
+
+// NOTE does not avoid allocations like the `BorrowedPlainMessage -> Message` conversion. this is
+// only used in (integration) tests
```

Comment:
Seems like `Message::try_from(PlainMessage)` is used in `first_handshake_message`, which is not just an integration test?

That seems OK (avoiding it seems to require quite a bit of change on the `Acceptor` API side) but this comment is potentially misleading.